packages/js/product-editor/src/components/attribute-control/attribute-table-row.tsx

```diff
@@ -344,50 +351,87 @@ export const AttributeTableRow: React.FC< AttributeTableRowProps > = ( {
 					suggestions={ tokenFieldSuggestions }
 					value={ tokenFieldValues }
 					onChange={ ( nextTokens: ( TokenItem | string )[] ) => {
+						// If there is no attribute, exit.
 						if ( ! attribute ) {
 							return;
 						}
 
-						// Create a new strings array with the new selected tokens,
-						const nextStringTokens =
-							nextTokens.map( tokenItemToString );
-
 						/*
-						 * Create an array with the new tokens to add,
-						 * filtering the new selected ones that are not in the
-						 * suggestions array.
+						 * Pick the new tokens from the nextTokens array.
+						 * (all new tokens are strings).
 						 */
-						const newTokens = nextStringTokens
-							.filter(
-								( t ) => ! tokenFieldSuggestions.includes( t )
-							)
-							.map( stringToTokenItem );
-
-						// Selected Terms to pass to the Form.
-						const selectedTerms = isGlobalAttribute
-							? nextStringTokens
-							: terms?.filter( ( term ) =>
-									nextStringTokens.includes( term.name )
-							  );
-
-						// Call the callback to update the Form terms.
-						onTermsSelect( selectedTerms, index, attribute );
+						const newTokens = nextTokens.filter(
+							( token ): token is string =>
+								typeof token === 'string'
+						);
+
+						// Create a list of the next string tokens.
+						const nextStringTokens =
+							nextTokens.map( tokenItemToString );
 
-						// If it is a global attribute, set the local terms.
+						// *** LOCAL Attributes ***
```

Comment:
This comment doesn't help me understand the code. In fact, it makes me more confused, since the very next line is

```ts
if ( isGlobalAttribute )
```

Reply:
You're right. Updated here https://github.com/woocommerce/woocommerce/pull/48627/commits/436e21abadd1e73036c5a7b85fa4e83827375a19